src/unix/epoll.rs

```diff
@@ -0,0 +1,429 @@
+    /// (data does not fit in a i32).
+    pub fn fd(&self) -> RawFd {
+        self.u64 as i32
+    }
+}
+
+/// Wrapper over epoll functionality.
+#[derive(Debug)]
+pub struct Epoll {
+    epoll_fd: RawFd,
+}
+
+impl Epoll {
+    /// Create a new epoll file descriptor.
+    pub fn new() -> io::Result<Self> {
+        let epoll_fd = SyscallReturnCode(unsafe { epoll_create1(EPOLL_CLOEXEC) }).into_result()?;
+        Ok(Epoll { epoll_fd })
+    }
+
+    /// Wrapper for `libc::epoll_ctl`.
+    ///
+    /// This can be used for adding, modifying or removing a file descriptor in the
+    /// interest list of the epoll instance.
+    ///
+    /// # Arguments
+    ///
+    /// * `operation` refers to the action to be performed on the file descriptor.
+    /// * `fd` is the file descriptor on which we want to perform `operation`.
+    /// * `event` refers to the `epoll_event` instance that is linked to `fd`.
+    pub fn ctl(
+        &self,
+        operation: ControlOperation,
+        fd: RawFd,
+        event: &EpollEvent,
+    ) -> io::Result<()> {
+        // Safe because we give a valid epoll file descriptor, a valid file descriptor to watch,
+        // as well as a valid epoll_event structure. We also check the return value.
+        SyscallReturnCode(unsafe {
+            epoll_ctl(
+                self.epoll_fd,
+                operation as i32,
+                fd,
+                event as *const EpollEvent as *mut epoll_event,
+            )
+        })
+        .into_empty_result()
+    }
+
+    /// Wrapper for `libc::epoll_wait`.
+    /// Returns the number of file descriptors in the interest list that became ready
+    /// for I/O or `errno` if an error occurred.
+    ///
+    /// # Arguments
+    ///
+    /// * `max_events` is the maximum number of events that we want to be returned in
+    /// `events` buffer.
+    /// * `timeout` specifies for how long the `epoll_wait` system call will block
+    /// (measured in milliseconds).
+    /// * `events` points to a memory area that will be used for storing the events
+    /// returned by `epoll_wait()` call.
+    pub fn wait(
+        &self,
+        max_events: usize,
+        timeout: i32,
+        events: &mut [EpollEvent],
+    ) -> io::Result<usize> {
+        // Safe because we give a valid epoll file descriptor and an array of epoll_event structures
+        // that will be modified by the kernel to indicate information about the subset of file
+        // descriptors in the interest list. We also check the return value.
+        let events_count = SyscallReturnCode(unsafe {
+            epoll_wait(
+                self.epoll_fd,
+                events.as_mut_ptr() as *mut epoll_event,
+                max_events as i32,
+                timeout,
+            )
+        })
+        .into_result()? as usize;
+
+        Ok(events_count)
+    }
+}
+
+impl AsRawFd for Epoll {
+    fn as_raw_fd(&self) -> RawFd {
+        self.epoll_fd
+    }
+}
+
+impl std::ops::Drop for Epoll {
```

Comment:
nit: `use std::ops::Drop` and skip the `std::ops` to be consistent with the other implementations

Reply:
:+1: